client/src/main/java/org/schemarepo/client/Avro1124RESTRepositoryClient.java

```diff
@@ -93,7 +93,7 @@ public Subject lookup(String subject) {
   public Iterable<Subject> subjects() {
     ArrayList<Subject> subjectList = new ArrayList<Subject>();
     try {
-      String subjects = webResource.get(String.class);
+      String subjects = webResource.accept(MediaType.TEXT_PLAIN_TYPE).get(String.class);
```

Comment:
(Unfortunately) it happens so that the default `Accept` value is
`text/html`. It's idiosyncrasy of jersey client.

Perhaps RESTRepository needs a new parameter to enable HTML support (so
that whoever needs strict compatibility with the original behavior can turn
it off). Any other thoughts?

On Tue, Jan 13, 2015 at 9:51 AM, Felix GV <notifications@github.com> wrote:

> In
> client/src/main/java/org/schemarepo/client/Avro1124RESTRepositoryClient.java
> <https://github.com/schema-repo/schema-repo/pull/35#discussion-diff-22879824>
> :
>
> > @@ -93,7 +93,7 @@ public Subject lookup(String subject) {
> >    public Iterable<Subject> subjects() {
> >      ArrayList<Subject> subjectList = new ArrayList<Subject>();
> >      try {
> > -      String subjects = webResource.get(String.class);
> > +      String subjects = webResource.accept(MediaType.TEXT_PLAIN_TYPE).get(String.class);
>
> This worries me a little bit...
>
> The intent of the Avro1124RESTRepositoryClient class is to have a test
> which emulates, as much as humanly possible, the behavior of the original
> client implementation in the AVRO-1124 ticket, so that folks using that
> code can upgrade to the schema-repo as seamlessly as possible.
>
> Adding this accept clause seems to imply that the default (or unspecified)
> media type sent by the Jersey client implementation would break against the
> new schema repo server code. If this is false, then can we just remove the
> accept(MediaType.TEXT_PLAIN_TYPE) statement? If this is true, then does
> that mean this PR introduces server-side changes that are incompatible with
> the old (AVRO-1124) clients, and if so, is there anything we can do to
> prevent or mitigate that?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/schema-repo/schema-repo/pull/35/files#r22879824>.
>